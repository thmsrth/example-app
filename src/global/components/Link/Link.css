.Link:hover {
  cursor: pointer;
}

.Link--disabled:hover {
  cursor: not-allowed;
}
