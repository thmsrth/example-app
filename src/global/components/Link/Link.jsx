import React from 'react';
import PropTypes from 'prop-types';

import { bemCls } from '../../utils/ClassNameHelpers';
import Icon from '../Icon';
import './Link.css';

const Link = ({
  className, href, text, target, disabled, onClick, iconName, tooltip,
}) => {
  if (href) {
    if (disabled) {
      return (
        <div className={`Link ${className} ${
          bemCls(`Link ${className}`, '--disabled', disabled)}`}
        >
          {iconName
          && (
          <div className={`${bemCls(`Link ${className}`, '__icon')} ${
            bemCls(`Link ${className}`, '__icon--disabled', disabled)}`}
          >
            <Icon name={iconName} />
          </div>
          )
        }
          {text
          && (
          <div className={`${bemCls(`Link ${className}`, '__text')} ${
            bemCls(`Link ${className}`, '__text--disabled', disabled)}`}
          >
            {text}
          </div>
          )
        }
        </div>
      );
    }
    return (
      <a
        className={`Link ${className}`}
        href={href}
        onClick={onClick}
        target={target}
        title={tooltip}
      >
        {iconName
          && (
          <div className={`${bemCls(`Link ${className}`, '__icon')} ${
            bemCls(`Link ${className}`, '__icon')}`}
          >
            <Icon name={iconName} />
          </div>
          )
        }
        {text
          && (
          <div className={`${bemCls(`Link ${className}`, '__text')} ${
            bemCls(`Link ${className}`, '__text')}`}
          >
            {text}
          </div>
          )
        }
      </a>
    );
  }
  return null;
};

export default Link;

Link.propTypes = {
  // children: PropTypes.element,
  className: PropTypes.string,
  text: PropTypes.string,
  href: PropTypes.string,
  target: PropTypes.string,
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  iconName: PropTypes.string,
  tooltip: PropTypes.string,
};

Link.defaultProps = {
  // children: null,
  className: '',
  iconName: '',
  text: '',
  href: undefined,
  target: '_self',
  disabled: false,
  onClick: () => {},
  tooltip: '',
};
