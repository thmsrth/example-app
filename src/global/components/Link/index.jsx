import React from 'react';
import PropTypes from 'prop-types';

import Link from './Link';

const LinkWrapper = ({ disabled, ...otherProps }) => {
  const handleOnClick = (e) => {
    if (disabled) e.preventDefault();
  };

  return (
    <Link
      {...otherProps}
      disabled={disabled}
      onClick={handleOnClick}
    />
  );
};

export default LinkWrapper;

LinkWrapper.propTypes = {
  disabled: PropTypes.bool,
};

LinkWrapper.defaultProps = {
  disabled: false,
};
