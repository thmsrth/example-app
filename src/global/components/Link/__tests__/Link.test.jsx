import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Link from '../Link';

const props = {
  className: 'myClassname',
  href: '/contact/user/400000227',
  text: 'Mytext',
};

describe('Link', () => {
  it('should render the component correctly with text and href', () => {
    const tree = shallow(<Link {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component correctly with children and href', () => {
    const component = (
      <Link {...props}>
        <div className="MyChildComponent" />
      </Link>
    );
    const tree = shallow(component);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component correctly with disabled state', () => {
    const tree = shallow(<Link {...props} disabled />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should not render the component if href is blank', () => {
    const tree = shallow(<Link
      {...props}
      href={undefined}
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
