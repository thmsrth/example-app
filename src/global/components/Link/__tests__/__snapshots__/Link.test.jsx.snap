// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should not render the component if href is blank 1`] = `""`;

exports[`Link should render the component correctly with children and href 1`] = `
<a
  className="Link myClassname"
  href="/contact/user/400000227"
  onClick={[Function]}
  target="_self"
  title=""
>
  <div
    className="Link__text myClassname__text Link__text myClassname__text"
  >
    Mytext
  </div>
</a>
`;

exports[`Link should render the component correctly with disabled state 1`] = `
<div
  className="Link myClassname Link--disabled myClassname--disabled"
>
  <div
    className="Link__text myClassname__text Link__text--disabled myClassname__text--disabled"
  >
    Mytext
  </div>
</div>
`;

exports[`Link should render the component correctly with text and href 1`] = `
<a
  className="Link myClassname"
  href="/contact/user/400000227"
  onClick={[Function]}
  target="_self"
  title=""
>
  <div
    className="Link__text myClassname__text Link__text myClassname__text"
  >
    Mytext
  </div>
</a>
`;
